### PR TITLE
added Iterators.reverse methods to support lazy reverse traversal

### PR DESCRIPTION
**Pull request by caseykneale**
Found this useful in a project (currently stubbing in the function calls now)... Sometimes the things at the bottom of the LRUCache are "interesting" and iterating in the typical sense is costly. Meanwhile reverse iteration is very much possible with the machinery in place. 

Let me know what you all think.

### CODE REVIEW

File: src/LRUCache.jl [hunk 1]
```diff
@@ -232,4 +233,23 @@ function _finalize_evictions!(finalizer, evictions)
     return
 end
 
+# Reverse iterator for LRUCache.LRU
+Base.eltype(::Type{Iterators.Reverse{LRUCache.LRU{T}}}) where {T} = eltype(T)
+Base.IteratorSize(::Type{Iterators.Reverse{LRUCache.LRU{T}}}) where {T} = Base.IteratorSize(LRUCache.CyclicOrderedSet)
+Base.IteratorEltype(::Type{Iterators.Reverse{LRUCache.LRU{T}}}) where {T} = Base.IteratorSize(LRUCache.LRU)
```
Comment (Jutho) [inline src/LRUCache.jl]: Right hand side contains a typo

File: src/LRUCache.jl [hunk 2]
```diff
@@ -232,4 +233,23 @@ function _finalize_evictions!(finalizer, evictions)
...
+Base.length(lru::Iterators.Reverse{LRUCache.LRU{T}}) where {T} = length(lru.keyset)
```
Comment (Jutho) [inline src/LRUCache.jl]: `lru` is the `Reverse` object, so its field should probably be `lru.itr`. Did you test this?

Comment (caseykneale) [reply src/LRUCache.jl]: Yes it was tested. See the unit tests added.

File: src/LRUCache.jl (hunk 1 above)
Comment (caseykneale) [reply src/LRUCache.jl]: Woops! I'll fix this later on today

File: src/LRUCache.jl (hunk 2 above)
Comment (caseykneale) [reply src/LRUCache.jl]: I'll fix this. I agree it should be `lru.itr.keyset`

**Comment by caseykneale** [comment]
Are these CI / Julia 1.0 errors due to changes in Pkg?

**Comment by Jutho** [comment]
My apologies for forgetting about this one. I would be happy to merge this; the only thing do be done I think is to remove the Manifest.toml file from the PR. Could you still do this, and then let's see what CI does and whether we should investigate the failures.

**Comment by Jutho** [comment]
I've moved this over to #33 , I think that you implemented many more methods than necessary, and that some of these are not correct. However, I very much appreciate that you initiated this, and thus plan to have support for `Iterators.reverse` in an upcoming version.

**Pull request closed without merging**